### ISSUE TITLE
[Feature Request] Auto-caption files with their names when uploading/forwarding

### ISSUE THREAD

**Issue by tissole** [kotatogram/kotatogram-desktop#176]
### Is your feature request related to a problem?
Uploaded files arrive at the destination without a caption, a big problem especially in the case of video files; the user must caption them one by one which is time-consuming. Same when forwarding files that do not have captions.

### Describe the solution you'd like
Add an option to auto-caption files with their names when uploading/forwarding. Is a very useful feature especially in the case of video files.

**Comment by ilya-fedin** [comment]
> the user must

Why must? Captions are optional.

**Comment by tissole** [comment]
Yes, optional. This feature could be turned on/off by the user. I was referring to what's happening now, if the user wants his files captioned he must go to the painful path to do it file by file :-(

**Comment by ilya-fedin** [comment]
Why user may want to caption each file?

**Comment by tissole** [comment]
If someone wants to check if already has a file who wants to forward is a bit difficult because TG does not let you copy the name of the file, you have to do save the file as...and copy the name and then search for the file. With caption is easier because can copy the text.

But the big problem is with videos. TG does not auto-caption uploaded videos and the user is left with a bunch of "anonymous" videos. Let's say someone uploads a course that has 50 videos. After he finishes he doesn't know how the videos are named because there is no visible name on the videos. Making the problem worse is the random way videos are uploaded. Which is first? From where to begin watching? The result is a mess. Videos lost their order and the user has to open them, look for the name, search for the first, and then for the second, etc...It is awful.

Auto-captioning will make watching videos much easier because the user does not have to open each video to search for the first. And if the option to upload them in the correct order (if the files are numbered for example) is implemented, the whole experience will become pleasant.

**Comment by mbnoimi** [comment]
+1
I've the same issue with tg. In addition to the mentioned above encoded mp4 files unsearchable so auto-caption (file name) will fix this issue.

**Comment by yo8z6gv** [comment]
+1
